### PR TITLE
Fixes exploit allowing players to open Bag of Holding rifts in the Heretic dimension

### PR DESCRIPTION
**Pull request by TheVekter**
## About The Pull Request

This PR prevents players from opening a BoH tear in the Heretic dimension areas. It also introduces a new area flag, "NO_BOH", which can be added to areas to replicate this behavior.

## Why It's Good For The Game

This prevents the (very unlikely) edge case that someone manages to open a BoH tear in one of the funny hand rooms, which is bad not only because it fucks over anyone else who gets sacrificed, but could also interfere with other things on that z-level (shuttles, other Lazy Template areas, etc). 

Messing with some new code here, so if there's better ways to handle this, by all means, chime in. This is a much more elegant method than just banning anything but the station.
## Changelog
:cl: Vekter
fix: Fixes an exploit that allowed players to open a Bag of Holding rift in the Heretic dimension.
/:cl:

### CODE REVIEW

File: code/datums/storage/subtypes/bag_of_holding.dm [hunk 1]
```diff
@@ -24,6 +25,7 @@
 		|| QDELETED(user) \
 		|| !user.can_perform_action(parent, NEED_DEXTERITY) \
 		|| !can_insert(to_insert, user) \
+		|| bag_area.area_flags & NO_BOH \
```
Comment (Kapu1178) [inline code/datums/storage/subtypes/bag_of_holding.dm:28]:
```suggestion
		|| (bag_area.area_flags & NO_BOH) \
```

**Comment by TheVekter** [comment]
Yes I tested this, it does the same thing every other instance of being unable to actually put the bags together does (nothing), I also tested it on the cargo shuttle and it blew up so we're good. I ideally would like to add some kind of flavorful feedback as to why it's not working but it's 1:20 AM here and I'm sick so fuck it.

File: code/datums/storage/subtypes/bag_of_holding.dm (hunk 1 above)
Comment (TheVekter) [reply code/datums/storage/subtypes/bag_of_holding.dm:28]: yeah okay fair

**Comment by DATA-xPUNGED** [comment]
> Yes I tested this, it does the same thing every other instance of being unable to actually put the bags together does (nothing), I also tested it on the cargo shuttle and it blew up so we're good. I ideally would like to add some kind of flavorful feedback as to why it's not working but it's 1:20 AM here and I'm sick so fuck it.

"Unknown Forces prevent the two bags from inching any closer."

**Comment by GPeckman** [comment]
You should probably explain why "BoH bombing in the heretic dimension" is actually a bad thing (That the resulting singulo can smash through other important places on the centcom z-level, like the nukie base) in your PR's body.

**Comment by TheVekter** [comment]
> You should probably explain why "BoH bombing in the heretic dimension" is actually a bad thing (That the resulting singulo can smash through other important places on the centcom z-level, like the nukie base) in your PR's body.

1) It's not the Centcom z-level, that's a different one. This is the one other Lazy Templates load on (nukie base, wizard's den, Bitrunning domains I think?).
2) I think it's generally bad at all because ideally our players shouldn't be able to just destroy all of the Heretic dimensions and fuck over anyone else who gets sacrificed. The primary reason isn't "it could fuck up other things on that z-level", it's "this shouldn't be possible".

**Pull request merged**